mimic/model/nova_objects.py

```diff
@@ -0,0 +1,392 @@
+                                             self.flavor_ref))}],
+                "rel": "bookmark"
+            },
+            "image": {
+                "id": self.image_ref,
+                "links": [{
+                    "href": absolutize_url("{0}/images/{1}".format(
+                        tenant_id, self.flavor_ref))
+                }]
+            }
+            if self.image_ref is not None else '',
+            "links": self.links_json(absolutize_url),
+            "metadata": self.metadata,
+            "name": self.server_name,
+            "tenant_id": tenant_id,
+            "status": self.status
+        })
+        return template
+
+    def creation_response_json(self, absolutize_url):
+        """
+        A JSON-serializable object returned for the initial creation of this
+        server.
+        """
+        return {
+            'server': {
+                "OS-DCF:diskConfig": self.disk_config,
+                "id": self.server_id,
+                "links": self.links_json(absolutize_url),
+                "adminPass": self.admin_password,
+            }
+        }
+
+    @classmethod
+    def from_creation_request_json(cls, collection, creation_json,
+                                   ipsegment=lambda: randrange(255)):
+        """
+        Create a :obj:`Server` from a JSON-serializable object that would be in
+        the body of a create server request.
+        """
+        now = collection.clock.seconds()
+        server_json = creation_json['server']
+        self = cls(
+            collection=collection,
+            server_name=server_json['name'],
+            server_id=('test-server{0}-id-{0}'
+                       .format(str(randrange(9999999999)))),
+            metadata=server_json.get("metadata") or {},
+            creation_time=now,
+            update_time=now,
+            private_ips=[
+                IPv4Address(address="10.180.{0}.{1}"
+                            .format(ipsegment(), ipsegment())),
+            ],
+            public_ips=[
+                IPv4Address(address="198.101.241.{0}".format(ipsegment())),
+                IPv6Address(address="2001:4800:780e:0510:d87b:9cbc:ff04:513a")
+            ],
+            creation_request_json=creation_json,
+            flavor_ref=server_json['flavorRef'],
+            image_ref=server_json['imageRef'] or '',
+            disk_config="AUTO",
+            # ^ TODO: should be: server_json['OS-DCF:diskConfig'],
+            status="ACTIVE",
+            admin_password="testpassword",
+            # ^ TODO: should be random
+        )
+        collection.servers.append(self)
+        return self
+
+
+@attributes(["address"])
+class IPv4Address(object):
+    """
+    An IPv6 address for a server.
+    """
+
+    def json(self):
+        """
+        A JSON-serializable representation of this address.
+        """
+        return {"addr": self.address, "version": 4}
+
+
+@attributes(["address"])
+class IPv6Address(object):
```

Comment:
I'd actually prefer to keep two separate classes, because we could add validation to the format for the address string itself at some point, and that differs between the two classes.